### DISCUSSION TITLE
Add support for background tasks

### DISCUSSION THREAD

**Discussion by beenje** [prefix-dev/pixi#1895] [Ideas]
### Problem description

I found 2 PRs attempting to add background tasks support: https://github.com/prefix-dev/pixi/pull/386 and https://github.com/prefix-dev/pixi/pull/682

Both were closed, as not complete and having problem supporting windows.

I'd really like this feature!
Opening this issue so it doesn't get forgotten and hoping someone will give it another try :-)

**Comment by ruben-arts** [comment]
Could you add a use-case and some more information why this is useful to you?

**Comment by ytausch** [comment]
I am also interested in this feature. My use case is: I want a `jupyter` task that runs jupyter in the background, and a `ui` task that starts our business UI (which also runs in the background). `ui` needs `jupyter` to be running, otherwise it will fail. With proper pixi integration, I would expect to be able to use `depends_on` for specifying that `jupyter` needs to be started before the UI can be run.

**Comment by wolfv** [comment]
For reference, the PR that @hadim made was really good and we could totally revive it.

**Comment by beenje** [comment]
To add my use case: I work a lot with [tango-controls](https://www.tango-controls.org), which requires a Tango database to be running. For local development, I'd like to run `pixi run tango-db` to start `Databaseds 2 -ORBendPoint giop:tcp:0.0.0.0:10000` in the background. It's then possible to register and run another tango server.

**Comment by baszalmstra** [comment]
I think this is an excellent idea but Im not sure when we'll get to it. For now I moved it to the ideas discussion board.

**Comment by jtroe** [comment]
I would also like this functionality, my use case is building Sphinx documentation, then building a Gatsby site which has a GraphQL API that reads the Sphinx JSON.

I have a task `pixi run docs:preview` which builds the JSON from the code and launches the site.  I would like to build then JSON, bootstrap watchdog (e.g. `watchmedo shell-command --patterns="*.py;*.rst' --command='make json'`), then build and preview the Gatsby site, which looks at the json output without caching when running on local.

This would be super helpful! (And I would be open to alternate suggestions also if you see an easier or better way)

Thanks!

**Comment by mtmal** [comment]
Hi all,

I’d like to bump this idea and add another concrete use case. In one of our robotics projects, we use pixi extensively. We structure the setup around two main features: ros2 and isaac. On top of these, we define two environments: robot, which depends only on the ros2 feature, and sim, which depends on both ros2 and isaac.

This allows us to launch ROS 2 applications with the appropriate flags, settings, etc., depending on whether they are running on the real robot or in simulation. However, in the simulation case, the simulator itself has to be started first via a separate task.

Being able to run simulations from a single task, one that would first start Isaac Sim, Gazebo, or a similar simulator, and then launch our application, would be very convenient. It would also significantly streamline demos; at the moment, we rely on custom shell scripts to start each application individually.

Thanks!
